crates/ark/src/lsp/symbols.rs

```diff
@@ -120,29 +120,73 @@ fn is_indexable(node: &Node) -> bool {
     true
 }
 
+// Function to parse a comment and return the section level and title
+fn parse_comment_as_section(comment: &str) -> Option<(usize, String)> {
+    // Match lines starting with one or more '#' followed by some non-empty content and must end with 4 or more '-' or '#'
+    // Ensure that there's actual content between the start and the trailing symbols.
+    let comment_re = regex::Regex::new(r"^(#+)\s*([^\s#-].+?)\s*(#{4,}|-{4,}|={4,})$").unwrap();
+
+    if let Some(caps) = comment_re.captures(comment) {
+        let hashes = caps.get(1)?.as_str().len();  // Count the number of '#'
+        let title = caps.get(2)?.as_str().trim().to_string();  // Extract the title text without trailing punctuations
+        return Some((hashes, title));  // Return the level based on the number of '#' and the title
+    }
+
+    None
+}
+
 fn index_node(
```

Comment:
Some tests are needed for this function too. You can do this to get a `Rope` and node:

```rs
let code = "my_code";
let document = Document::new(code, None);
let node = ast.root_node();
```